tests/Samples.Server.Tests/BaseTest.cs

```diff
@@ -46,23 +39,46 @@ protected async Task<string> SendRequestAsync(string text)
             return await response.Content.ReadAsStringAsync();
         }
 
-        protected async Task<string> SendRequestAsync(GraphQLRequest request)
+        protected async Task<string> SendRequestAsync(GraphQLRequest request, RequestType requestType, string requestUri = "graphql")
         {
-            var response = await Client.PostAsync("graphql", new StringContent(JsonConvert.SerializeObject(request), Encoding.UTF8, "application/json"));
-            return  await response.Content.ReadAsStringAsync();
+            // Different servings over HTTP:
+            // https://graphql.org/learn/serving-over-http/
+            HttpResponseMessage response;
+            switch (requestType)
+            {
+                case RequestType.Get:
+                    var queryString = await Serializer.ToQueryStringParamsAsync(request);
+                    var url = $"{requestUri}?{queryString}";
+                    response = await Client.GetAsync(url);
+                    break;
+                case RequestType.PostWithJson:
+                    var jsonContent = Serializer.ToJson(request);
+                    response = await Client.PostAsync(requestUri, new StringContent(jsonContent, Encoding.UTF8, MediaType.Json));
+                    break;
+                case RequestType.PostWithGraph:
+                    response = await Client.PostAsync(requestUri, new StringContent(request.Query, Encoding.UTF8, MediaType.GraphQL));
```

Comment:
Not quite right. In this case, the variables and operation name should be passed in request query. The same for PostWithForm.

Reply:
I saw that you set query on the caller side, but in the case of GET this happens inside the method. Why so?

Reply:
For PostWithGraph, 
* Correct, I should support the variables and operation name in the querystring params.

For PostWithForm,
* I believe the body can contain all three request pieces, but you're right in that a caller *could* also supply querystring params too, so it should support both and support overriding by querystring params.

> I saw that you set query on the caller side, but in the case of GET this happens inside the method. Why so?
This was to test a specific case whereby the standard functionality (passing the request object + request type) is overridden by also passing a different set of values in the querystring, which should override the request body values.

I'll try make this clearer, probably need to separate the tests out a bit.

Reply:
I've made the send method a bit clearer, it now takes another `GraphQLRequest` object in an optional parameter `queryStringOverride` that is used by the test that checks that query string params are extracted out and take priority of anything in the body. 

Makes it clearer than the querystring `string` override I had, so it's clearer I was testing already exactly what you were asking for.

Reply:
excellent